zsh-syntax-highlighting.zsh

```diff
@@ -358,6 +358,16 @@ _zsh_highlight_buffer_modified()
   [[ "${_ZSH_HIGHLIGHT_PRIOR_BUFFER:-}" != "$BUFFER" ]]
 }
 
+# Whether zsh is done processing the input queue.
+# Note that for many highlighters there is no need parse the
+# current buffer state if there are still keys to be processed.
+#
+# Returns 0 if there are no more keys to be processed, ie, the buffer likely won't change for a while.
+_zsh_highlight_input_queue_empty()
+{
+    [[ ${KEYS_QUEUED_COUNT:-0} == 0 ]]
```

Comment:
I presume the `:-0` is so the tests run under `WARN_CREATE_GLOBAL`. Since `KEYS_QUEUED_COUNT` is a ZLE variable, I think it'd make sense to spoof it in the test harness (tests/test-highlighting.zsh in the `run_test_internal` function) along with `CURSOR` and `PENDING`.

Since the variable is numeric, you could use an arithmetic expression (e.g. `(( KEYS_QUEUED_COUNT == 0 ))`), but that's just style.

Reply:
Ah interesting, I just used `${KEYS_QUEUED_COUNT:-0}` for "good measure" in case it's not set in some older version or for some other reason. With that default it should always behave reasonably. But it looks like I circumvented testing a bit with this.

It makes sense to spoof it. Should we rely on the variable to exist? I'm not sure if `zsh-syntax-highlighting` is using `set -u` overerall.